docs/framework/wpf/introduction-to-wpf.md

```diff
@@ -0,0 +1,664 @@
+[!code-csharp[IntroToWPFSnippets#HandleEllipseMouseUpEventCODEBEHIND](~/samples/snippets/csharp/wpf/introduction-to-wpf/introduction-to-wpf_8.cs)]
+
+The following figure shows what is produced by the preceding code:
+
+![A window with the text "you clicked the ellipse&#33;"](media/introduction-to-wpf/wpfintrofigure12.png)
+
+For more information, see [Shapes and basic drawing in WPF overview](../../desktop-wpf/data/data-binding-overview.md).
+
+### 2D geometries
+
+The 2D shapes provided by WPF cover the standard set of basic shapes. However, you may need to create custom shapes to facilitate the design of a customized user interface. For this purpose, WPF provides geometries. The following figure demonstrates the use of geometries to create a custom shape that can be drawn directly, used as a brush, or used to clip other shapes and controls.
+
+<xref:System.Windows.Shapes.Path> objects can be used to draw closed or open shapes, multiple shapes, and even curved shapes.
+
+<xref:System.Windows.Media.Geometry> objects can be used for clipping, hit-testing, and rendering 2D graphic data.
+
+![Various uses of a path](media/introduction-to-wpf/wpfintrofigure5.png)
+
+For more information, see [Geometry overview](graphics-multimedia/geometry-overview.md).
+
+### 2D effects
+
+A subset of WPF 2D capabilities includes visual effects, such as gradients, bitmaps, drawings, painting with videos, rotation, scaling, and skewing. These are all achieved with brushes; the following figure shows some examples:
+
+![Illustration of different brushes](media/introduction-to-wpf/wpfintrofigure6.png)
+
+For more information, see [WPF brushes overview](graphics-multimedia/wpf-brushes-overview.md).
+
+### 3D rendering
+
+WPF also includes 3D rendering capabilities that integrate with 2-d graphics to allow the creation of more exciting and interesting user interfaces. For example, the following figure shows 2D images rendered onto 3D shapes.
+
+![Visual3D sample screen shot](media/introduction-to-wpf/wpfintrofigure13.png)
+
+For more information, see [3D graphics overview](graphics-multimedia/3-d-graphics-overview.md).
+
+## Animation
+
+WPF animation support lets you make controls grow, shake, spin, and fade, to create interesting page transitions, and more. You can animate most WPF classes, even custom classes. The following figure shows a simple animation in action:
+
+![Images of an animated cube](media/introduction-to-wpf/wpfintrofigure7.png)
+
+For more information, see [Animation overview](graphics-multimedia/animation-overview.md).
+
+## Media
+
+One way to convey rich content is through the use of audiovisual media. WPF provides special support for images, video, and audio.
+
+### Images
+
+Images are common to most applications, and WPF provides several ways to use them. The following figure shows a user interface with a list box that contains thumbnail images. When a thumbnail is selected, the image is shown full-size.
```

Comment:
```suggestion
Images are common to most applications, and WPF provides several ways to use them. The following figure shows a user interface with a list box that contains thumbnail images. When a thumbnail is selected, the image is shown full-size:
```

Reply:
Not sure I agree with this one, so I'm going to leave it as-is.